packages/ui/src/core/check-box-group/check-box-group.component.tsx

```diff
@@ -1,19 +1,56 @@
-import React from 'react';
+import React, { useState } from 'react';
 import PT from 'prop-types';
 
 import { intrinsicComponent, objectValues } from '../../utils/functions';
 import CheckBox from '../check-box';
 import type { CheckBoxGroupProps } from './check-box-group.props';
 import Styled from './check-box-group.styles';
 import { LabelPosition } from './types';
+import { Size } from '../check-box/size';
+import CheckBoxGroupWithIcon from './check-box-group-with-icon.component';
 
 const CheckBoxGroup = intrinsicComponent<CheckBoxGroupProps, HTMLLabelElement>(
-  ({ label, checkBoxProps, readOnly, disabled, labelPosition, ...rest }: CheckBoxGroupProps, ref): JSX.Element => {
+  (
+    {
+      size = Size.Sm,
+      label,
+      checkBoxProps,
+      readOnly,
+      disabled,
+      labelPosition,
+      withIcon = false,
+      title,
+      icon,
+      ...rest
+    }: CheckBoxGroupProps,
+    ref
+  ): JSX.Element => {
+    const [isHovering, setIsHovering] = useState(false);
+
+    const handleEnteringTooltip = (): void => {
+      setTimeout(() => {
+        setIsHovering(true);
+      }, 250);
+    };
+
+    const handleLeavingTooltip = (): void => {
+      setTimeout(() => {
+        setIsHovering(false);
+      }, 250);
+    };
```

Comment:
All this part should be in checkbox itself not in checboxGroup. To not use internal props and also now the hover functionality doesn't work with checkbox only